### ISSUE TITLE
Invalid integrity attribute value when used with `laravel-mix-sri`

### ISSUE THREAD

**Issue by mbabker** [Elhebert/laravel-sri#77]
So I honestly have no idea if this is a bug with the Laravel framework integration or the manifest generated by the Mix plugin, so I'm logging it here for now because the PHP class seems to have an inconsistent behavior.

The integrity attribute should have a value such as `integrity="sha384-<hash>"`, however, the `Elhebert\SubresourceIntegrity\Sri` class will generate `integrity="<hash>"` when a resource is found in a Mix manifest, and this causes the browser to not be able to process the attribute correctly.

The Mix plugin writes `mix-sri.json` without the algorithm prefix that the attribute requires, so when the `Sri` class reads from it, it's not prepending the algorithm to the hash in the JSON file.  Compared to when the hash is generated at runtime, it does return a correctly prepended algorithm to the hash.

**Comment by akhan619** [comment]
Same issue here. There are obviously two solutions to this.
- Make a change to how the mix-sri.json is generated with the algorithm type attached.
- Or attach the algorithm string when generating the tags.

As far as I can see a simple change of line 61 in Sri.php from 

```return $json->$prefixedPath;```

 to 

```return "{$this->algorithm}-{$json->$prefixedPath}";```

Will do the trick. Of course, this might break compatibility with other file hash generation packages, not to mention that the right algorithm type also needs to be specified. So I do see why the author didn't implement it. In my opinion, the change should NOT be made considering that a SRI hash generator should include the algorithm in the hash as required by SRI. So the change should be made in the ```laravel-mix-sri``` package, since the ```laravel-sri``` package correctlys attaches the algorithm to the hash that it generates. The problem occurs when it uses a seperate ```mix-sri.json```, where it correctly assumes that the hash should be complete.

**Comment by akhan619** [comment]
Just played around with the ```laravel-mix-sri``` package as well. If we change line 35 in ```src/SriPlugin.ts``` from

```hashes[filePath] = crypto```

to

```hashes[filePath] = this.algorithm + '-' + crypto```

problem solved as well. Ofcourse, if you are doing it yourself you will need to build it. First,
- Install typescript
- Then ```cd node_modules/laravel-mix-sri```
- Run ```tsc --build tsconfig.json```

Note: These are all hacks and will break when you make updates. It's a very bad idea to change vendor or node_modules files directly. Better for the package author to implement, since there might be things I have overlooked and they will have a better picture of how the package works and should be changed.

With that very important warning out of the way, I would suggest wait for the author to update. If you can't wait and the package license permits (too lazy to check right now myself :) ) just fork the package and build your own. Of course, maintaining and fixing issues will be on you. Hope this helps anyone with the same issues.

**Comment by akhan619** [comment]
Ok, the local build options won't work if you are on Laravel Vapor, since vapor will pull the npm package. A quick and dirty hack for now is to use both the packages as is without making any changes. Just add the following to the ```subresource-integrity.php``` config file

```
'hashes' =>  array_combine(
        array_map(function($item) { return substr($item, 1);}, array_keys(json_decode(file_get_contents(public_path('mix-sri.json')), true))), 
        array_values(array_map(function($item) {return env('SRI_ALGORITHM', 'sha256') .  '-' . $item;}, json_decode(file_get_contents(public_path('mix-sri.json')), true)))
    ),
```
The above scary code is needed since the ```laravel-mix-sri``` generates a json which has a leading slash ```/``` while the hashes config value expects a key value for the array which doesn't have the slash. So we can enjoy the dirty hack above. Cheers!!!

**Comment by Elhebert** [comment]
Sorry for not responding sooner, I've been taking some time off.

As @akhan619 mentioned it ti problems comes from the npm package `laravel-mix-sri`. A PR was created and I merged it. I can only publish the new version to npm on Wednesday.

**Comment by Elhebert** [comment]
I just released v1.0.4 of laravel-mix-sri. Which fixes this bug.
